src/components/reservation/helper.ts

```diff
@@ -9,4 +10,37 @@ export const getPreferredBranch = (id: string, array: AgencyBranch[]) => {
   return locationItem ? locationItem.title : id;
 };
 
+export const hardcodedInterestPeriods = (t: UseTextFunction) => {
+  return {
+    "30": t("oneMonthText"),
+    "60": t("twoMonthsText"),
+    "90": t("threeMonthsText"),
+    "180": t("sixMonthsText"),
+    "360": t("oneYearText")
+  };
+};
+
+export const getNoInterestAfter = (
+  days: number | string,
+  t: UseTextFunction
+) => {
+  const reservationInterestIntervals: { [key: string]: string } = {
+    ...hardcodedInterestPeriods(t),
+    default: `${days} ${t("daysText")}`
+  } as const;
+
+  const lookupKey = String(days);
+  return (
+    reservationInterestIntervals[lookupKey] ??
+    reservationInterestIntervals.default
+  );
+};
+
+export const getFutureDateString = (num: string | number) => {
+  const today = new Date();
+  const futureDate = new Date();
+  futureDate.setDate(today.getDate() + Number(num));
+  return futureDate.toISOString().slice(0, 10);
+};
```

Comment:
Please rework this function.

Our internal representation of the users interest date should be a number of days wherever possible. Not a string.

When working with dates where the standard JavaScript date api does not help us much (outputting a date in a specific format, modifying a date) we use `Day.js`. It provides helpful methods e.g. [`format()`] and [`add()'](https://day.js.org/docs/en/manipulate/add).